debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlDdlParser.java

```diff
@@ -690,10 +697,26 @@ protected void parseColumnDefinition(Marker start, String columnName, TokenStrea
         column.jdbcType(dataType.jdbcType());
         column.type(dataType.name(), dataType.expression());
         if ("ENUM".equals(dataType.name())) {
-            column.length(1);
+            String options = parseSetAndEnumOptions(dataType.expression());
+            //After DBZ-132, enum can be multi character
+            int maxLength = 0;
+            int enumLen = 0;
+            int optionLen = options.length();
+            for (int i=0;i<optionLen;i++) {
+                if (options.charAt(i) == MySqlDdlParser.ENUM_AND_SET_DELIMINATOR) {
+                    if (enumLen > maxLength) {
+                        maxLength = enumLen;
+                    }
+                    enumLen = 0;
+                } else {
+                    enumLen ++;
+                }
+            }
+            column.length(maxLength);
```

Comment:
As enum column will always has single enum value. Max column length should be length of longest string in allowed enum values, don't you  think ? 

List<String> options = parseSetAndEnumOptions(dataType.expression());
column.length(options.size());

This will give you total number of allowed enum values